src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -146,21 +150,28 @@ public X509Certificate[] getAcceptedIssuers() {
     @Provides
     @Singleton
     public static HttpClient provideHttpClient(
-            ScheduledExecutorService executor,
             SSLContext sslContext,
+            @Named(ConfigModule.CRYOSTAT_AGENT_AUTHORIZATION) String authorization,
             @Named(ConfigModule.CRYOSTAT_AGENT_WEBCLIENT_SSL_VERIFY_HOSTNAME)
-                    boolean verifyHostname,
-            @Named(ConfigModule.CRYOSTAT_AGENT_WEBCLIENT_CONNECT_TIMEOUT_MS)
-                    long connectTimeoutMs) {
-        System.getProperties()
-                .setProperty(
-                        "jdk.internal.httpclient.disableHostnameVerification",
-                        Boolean.toString(!verifyHostname));
-        return HttpClient.newBuilder()
-                .executor(executor)
-                .connectTimeout(Duration.ofMillis(connectTimeoutMs))
-                .sslContext(sslContext)
-                .build();
+                    boolean verifyHostname) {
+        HttpClientBuilder builder =
+                HttpClients.custom()
+                        .setDefaultHeaders(Set.of(new BasicHeader("Authorization", authorization)))
+                        .setSSLContext(sslContext)
+                        .setDefaultRequestConfig(
+                                RequestConfig.custom()
+                                        .setAuthenticationEnabled(true)
+                                        .setExpectContinueEnabled(true)
+                                        .setConnectTimeout(5000)
```

Comment:
Should these timeout values taken from envs (e.g. `CRYOSTAT_AGENT_WEBCLIENT_CONNECT_TIMEOUT_MS`)?

Reply:
Oh good catch, thanks. I didn't catch on to that when I was rebasing this work.